### PR TITLE
[5.2] Static function getList deprecated in mod_stats

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Static function getList is after applying pull request deprecated and new non-static method getStats created in StatsHelper.

### Testing Instructions

1. Make sure you have site module mod_stats published
2. You should see module output on frontend
3. Deprecated static function getList should also work

mod_stats

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
After click on ```Apply Patch``` in Patchtester only a red Message is shown: "The file marked for modification does not exist: modules/mod_stats/src/Dispatcher/Dispatcher.php".

**Comment by brianteeman** [comment]
@fgsw Your local joomla installation is not up to date

**Comment by fgsw** [comment]
My Installation Setting:
image

image

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 15ba9800d2b8f6ef5a57e2b944de4919e3002866<br /><br />Using nightly build.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by ceford** [comment]
Just realised I am testing with PHP8.3 - everything seems to work. For this PR I do not see a deprecated notice with or without the patch. Am I supposed to?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by joomlaweby** [comment]
> Just realised I am testing with PHP8.3 - everything seems to work. For this PR I do not see a deprecated notice with or without the patch. Am I supposed to?
> 
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/42907](https://issues.joomla.org/tracker/joomla-cms/42907).

@ceford  No, deprecated static function in helper file can be still used by other extensions (probably only 3rd party), so this is the case, when this deprecated function is called. Module mod_stats is now using new non-static method from helper, so there is no deprecation notice.

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 15ba9800d2b8f6ef5a57e2b944de4919e3002866<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

File: modules/mod_stats/src/Helper/StatsHelper.php [hunk 1]
```diff
@@ -25,19 +27,23 @@
...
      * @param   \Joomla\Registry\Registry  &$params  module parameters
      *
      * @return  array
+     *
+     * @since   __DEPLOY_VERSION__
      */
-    public static function &getList(&$params)
+    public function getStats(&$params)
     {
         $app        = Factory::getApplication();
```
Comment (laoneo) [inline modules/mod_stats/src/Helper/StatsHelper.php]: Can you also inject here the $app object like in the others?

Comment (joomlaweby) [reply modules/mod_stats/src/Helper/StatsHelper.php]: Done. Is it OK now?

File: modules/mod_stats/src/Helper/StatsHelper.php [hunk 2]
```diff
@@ -25,19 +29,23 @@
...
-     * @param   \Joomla\Registry\Registry  &$params  module parameters
+     * @param   Registry  &$params  module parameters
+     * @param   CMSApplication  $app  The application
      *
      * @return  array
+     *
+     * @since   __DEPLOY_VERSION__
      */
-    public static function &getList(&$params)
+    public function getStats(Registry &$params, CMSApplication $app)
```
Comment (laoneo) [inline modules/mod_stats/src/Helper/StatsHelper.php]: What happens when change this here to CMSApplicationInterface?

Comment (joomlaweby) [reply modules/mod_stats/src/Helper/StatsHelper.php]: No problem, done! :-)

**Comment by richard67** [comment]
Back to pending as there have been made changes.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by exlemor** [comment]
I have tested this item :white_check_mark: successfully on 1dbe39ef2cd8e024617863220aaa094ed0863602<br /><br />Was successful and noticed that in Joomla Debug, there seems to be fewer Queries after Patch.

Before Patch: 28 statements were executed, 6 of which were duplicates, 22 unique
After Patch: 25 statements were executed, 2 of which were duplicates, 23 unique

so if intended great and even if not, fewer = better so great :)

Thanks for yet another good fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 1dbe39ef2cd8e024617863220aaa094ed0863602<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by cybersalt** [comment]
I have tested this item :white_check_mark: successfully on 1dbe39ef2cd8e024617863220aaa094ed0863602<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42907">issues.joomla.org/tracker/joomla-cms/42907</a>.</sub>

**Comment by brianteeman** [comment]
great to see all these tests @cybersalt - hope this will become a regular thing

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.2-dev.

**Comment by pe7er** [comment]
Thanks @joomlaweby !

**Pull request merged**